### PR TITLE
Evan points and end screen

### PR DESCRIPTION
**Pull request by Evan-Kreutzwiser**
Settlement upgrading and use quicksort for recursion marks

### CODE REVIEW

**Comment by Evan-Kreutzwiser** [comment]
Fixed the merge conflicts

File: KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java [hunk 1]
```diff
@@ -784,25 +852,53 @@ else if (cards[currentPlayer].get(i) == type) {
...
+     * Sort a player's list of cards using a recursive quick sort algorithm
      * @param player The player who's cards will be sorted
+     * @param left The left bounds of the segment to sort (used to recursion, set to 0)
+     * @param right The right bounds of the segment to sort (used to recursion, set to length of array - 1)
+     * @return 
      */
-    public void insertionSort(int player) {
-
-        // Get the player's card list
+    private void quickSortCards(int player, int left, int right) {
```
Comment (Tacitor) [inline KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:861]: Quik sort is spelt wrong but I don't feel like changing it so it's fine.

Comment (Evan-Kreutzwiser) [reply KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:861]: ive always spelt it with a c
look at the image in the chat

Comment (Tacitor) [reply KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:861]: nvm I'm just being smooth.

**Review by Tacitor — APPROVED** [review]
All good chief. Hopefully the last merge.

**Pull request merged**